assets/js/atcb.js

```diff
@@ -64,79 +70,75 @@
       // move entries one level up, but skip schema types
-      if (key.charAt(0) !== '@') {
-        atcbConfig[key] = atcbConfig['event'][key]; 
-      } 
+      if (key.charAt(0) !== "@") {
+        atcbConfig[key] = atcbConfig.event[key];
+      }
     });
     // drop the event block and return
     delete atcbConfig.event;
-  }
-  catch(err) {
-    console.error("add-to-calendar button problem: it seems like you use the schema.org style, but did not define it properly");
+  } catch (err) {
+    console.error(
+      "add-to-calendar button problem: it seems like you use the schema.org style, but did not define it properly"
+    );
   }
   return atcbConfig;
 }
 
-
-
 // BACKWARDS COMPATIBILITY REWRITE - you can remove this, if you did not use this script before v1.4.0.
 function atcb_patch_config(atcbConfig) {
   const keyChanges = {
-    'title':      'name',
-    'dateStart':  'startDate',
-    'dateEnd':    'endDate',
-    'timeStart':  'startTime',
-    'timeEnd':    'endTime',
+    title: "name",
+    dateStart: "startDate",
+    dateEnd: "endDate",
+    timeStart: "startTime",
+    timeEnd: "endTime",
   };
-  Object.keys(keyChanges).forEach(key => {
+  Object.keys(keyChanges).forEach((key) => {
     if (atcbConfig[keyChanges[key]] == null && atcbConfig[key] != null) {
       atcbConfig[keyChanges[key]] = atcbConfig[key];
     }
   });
   return atcbConfig;
 }
 
-
-
 // CLEAN DATA BEFORE FURTHER VALIDATION (CONSIDERING SPECIAL RULES AND SCHEMES)
 function atcb_decorate_data(atcbConfig) {
   // cleanup different date-time formats
   atcbConfig = atcb_date_cleanup(atcbConfig);
   // calculate the real date values in case that there are some special rules included (e.g. adding days dynamically)
-  atcbConfig['startDate'] = atcb_date_calculation(atcbConfig['startDate']);
-  atcbConfig['endDate'] = atcb_date_calculation(atcbConfig['endDate']);
-  
+  atcbConfig.startDate = atcb_date_calculation(atcbConfig.startDate);
+  atcbConfig.endDate = atcb_date_calculation(atcbConfig.endDate);
+
   // if no description or already decorated, return early
   if (!atcbConfig.description || atcbConfig.description_iCal) return atcbConfig;
 
   // make a copy of the given argument rather than mutating in place
   const data = Object.assign({}, atcbConfig);
   // standardize any line breaks in the description and transform URLs (but keep a clean copy without the URL magic for iCal)
-  data.description = data.description.replace(/<br\s*\/?>/gmi, '\n');
-  data.description_iCal = data.description.replace('[url]','').replace('[/url]','');
-  data.description = data.description.replace(/\[url\](.*?)\[\/url\]/g, "<a href='$1' target='_blank' rel='noopener'>$1</a>");
-  return data
+  data.description = data.description.replace(/<br\s*\/?>/gi, "\n");
+  data.description_iCal = data.description.replace("[url]", "").replace("[/url]", "");
+  data.description = data.description.replace(
+    /\[url\]([\w&$+.,:;=~!*'?@#\s\-()|/^%]*?)\[\/url\]/g,
```

Comment:
Codacy has a fix for the issue: [Unexpected non-greedy constant quantifier. The quantifier is effectively possessive, so it doesn't matter whether it is greedy or not.](https://app.codacy.com/gh/jekuer/add-to-calendar-button/pullRequest?prid=9707311)
```suggestion
    /\[url\]([\w&$+.,:;=~!*'?@#\s\-()|/^%]*)\[\/url\]/g,
```